Index: core_impl/com/itecheasy/core/fba/ReplenishmentOrderServiceImpl.java
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- core_impl/com/itecheasy/core/fba/ReplenishmentOrderServiceImpl.java	(date 1533115550082)
+++ core_impl/com/itecheasy/core/fba/ReplenishmentOrderServiceImpl.java	(date 1533115550082)
@@ -1062,10 +1062,20 @@
 
 	@Override
 	public void addAndUpdateOrderProductList(int orderId, List<UpdateOrderProductForm> orderProducts, String operator){
-
+		ReplenishmentOrderPO po = replenishmentOrderDao.getObject(orderId);
+		for (UpdateOrderProductForm form : orderProducts) {
+			if (form.getUpdateProductCode() != null) {
+				if (form.getOrderProductId()==0) {
+					ShopProductRelated relatedByOtherCode = fbaShopProductService.getRelatedByOtherCode(form.getUpdateProductCode(), po.getShopId());
+					form.setUnitQty(relatedByOtherCode.getUnitQuantity());
+					form.setSku(relatedByOtherCode.getSku());
+					form.setFbaShopProductId(relatedByOtherCode.getId());
+//                    cmsSkuList.add(form.getUpdateProductCode());
+				}
+			}
+		}
 
-
-		addOrderProducts();
+		addOrderProducts(orderId,orderProducts,operator);
 	}
 
 
